tensorflow_model_optimization/python/core/sparsity_tf2/pruner_test.py

```diff
@@ -0,0 +1,493 @@
+  #   # NOTE: save_round > start = 0, should not prune until save round is passed
+
+  #   self.initialize()
+    
+  #   p = pruner.LTHPruner(
+  #       pruning_schedule=pruning_schedule,
+  #       save_schedule=self.save_init,
+  #       block_size=self.block_size,
+  #       block_pooling_type=self.block_pooling_type)
+
+  #   optimizer = self.dummy_optimizer
+  #   optimizer.iterations.assign(tf.cast(self.training_step_fn(), tf.int64))
+  #   expected_saved_og_initialization = None
+
+  #   p.create_slots(optimizer, initial_weight)
+  #   for i in range(9): # this should save and not update, assumes pruning works correctly as per above tests
+  #     # perturb weights
+  #     initial_weight = tf.math.add(initial_weight, sample_noise(i))
+  #     p.prune(optimizer, initial_weight, self.grad(initial_weight))
+  #     if i == save_round:
+  #       expected_saved_og_initialization = optimizer.get_slot(initial_weight, "original_initialization").read_value()
+  #     optimizer.iterations.assign(tf.Variable(i))
+
+  #   og_initialization_slot = optimizer.get_slot(initial_weight, "original_initialization")
+  #   self.assertAllEqual(og_initialization_slot, expected_saved_og_initialization)
+
+  #   mask_after_pruning = K.get_value(optimizer.get_slot(initial_weight, "mask"))
+  #   self.assertAllEqual(np.count_nonzero(mask_after_pruning), 100)
+
+
+  # def testReloadWeightsatInit(self):
+  #   initial_weight = tf.Variable(np.linspace(1.0, 100.0, 100), name="weights")
+  #   weight_dtype = initial_weight.dtype.base_dtype
+
+  #   save_round = 0
+  #   n_rounds = 20
+  #   end_itr = 100
+  #   frequency, prune_pcent_per_round = get_lth_sparsity(save_round, n_rounds, self.target_sparsity, end_itr)
+  #   pruning_schedule = make_pruning_schedule(prune_pcent_per_round, save_round, end_itr, frequency)
+  #   # NOTE: save_round > start = 0, should not prune until save round is passed
+
+  #   self.initialize()
+    
+  #   p = pruner.LTHPruner(
+  #       pruning_schedule=pruning_schedule,
+  #       save_schedule=self.save_init,
+  #       block_size=self.block_size,
+  #       block_pooling_type=self.block_pooling_type)
+
+  #   optimizer = self.dummy_optimizer
+  #   optimizer.iterations.assign(tf.cast(self.training_step_fn(), tf.int64))
+  #   expected_saved_og_initialization = None
+
+  #   p.create_slots(optimizer, initial_weight)
+  #   for i in range(5): # this should save, reload, and update once
+  #     initial_weight = tf.math.add(initial_weight, sample_noise(i))
+  #     p.prune(optimizer, initial_weight, self.grad(initial_weight))
+  #     if i == save_round:
+  #       expected_saved_og_initialization = optimizer.get_slot(initial_weight, "original_initialization")
+  #     optimizer.iterations.assign(tf.Variable(i)) # save weights right before iteration
+
+  #   og_initialization_slot = optimizer.get_slot(initial_weight, "original_initialization")
```

Comment:
Rename all stuff test python vars w/
`og_initialization` to just `initialization`